src/main/java/io/confluent/connect/jdbc/validation/JdbcSourceConnectorValidation.java

```diff
@@ -150,6 +152,16 @@ private boolean validateMultiConfigs() {
     return true;
   }
 
+  private boolean hasAnyQueryConfig() {
+    String query = config.getString(JdbcSourceConnectorConfig.QUERY_CONFIG);
+    Password queryMasked =
+        config.getPassword(JdbcSourceConnectorConfig.QUERY_MASKED_CONFIG);
```

Comment:
If you make `config.getQuery` return optional, then this `hasAnyQueryConfig` method will not be needed.